driver/src/main/com/mongodb/reactivestreams/client/internal/GridFSAsyncStreamHelper.java

```diff
@@ -0,0 +1,207 @@
+                        callback.onResult(result, error);
+                    }
+                });
+            }
+
+            @Override
+            public void close(final SingleResultCallback<Void> callback) {
+                wrapped.close().subscribe(new Subscriber<Success>() {
+                    private Throwable error = null;
+
+                    @Override
+                    public void onSubscribe(final Subscription s) {
+                        s.request(1);
+                    }
+
+                    @Override
+                    public void onNext(final Success success) {
+                    }
+
+                    @Override
+                    public void onError(final Throwable t) {
+                        error = t;
+                    }
+
+                    @Override
+                    public void onComplete() {
+                        callback.onResult(null, error);
+                    }
+                });
+            }
+        };
+    }
+
+    static com.mongodb.async.client.gridfs.AsyncOutputStream toCallbackAsyncOutputStream(final AsyncOutputStream wrapped) {
+        return new com.mongodb.async.client.gridfs.AsyncOutputStream() {
+
+            @Override
+            public void write(final ByteBuffer src, final SingleResultCallback<Integer> callback) {
+                wrapped.write(src).subscribe(new Subscriber<Integer>() {
+                    private Integer result = null;
+                    private Throwable error = null;
+
+                    @Override
+                    public void onSubscribe(final Subscription s) {
+                        s.request(1);
+                    }
+
+                    @Override
+                    public void onNext(final Integer integer) {
+                        result = integer;
+                    }
+
+                    @Override
+                    public void onError(final Throwable t) {
+                        error = t;
+                    }
+
+                    @Override
+                    public void onComplete() {
+                        callback.onResult(result, error);
+                    }
+                });
+            }
+
+            @Override
+            public void close(final SingleResultCallback<Void> callback) {
+                wrapped.close().subscribe(new Subscriber<Success>() {
+                    private Throwable error = null;
+
+                    @Override
+                    public void onSubscribe(final Subscription s) {
+                        s.request(1);
+                    }
+
+                    @Override
+                    public void onNext(final Success success) {
+                    }
+
+                    @Override
+                    public void onError(final Throwable t) {
+                        error = t;
+                    }
+
+                    @Override
+                    public void onComplete() {
+                        callback.onResult(null, error);
```

Comment:
Any way to test this?  I don't see it invoked when running the tests in the debugger.